railgun/internal/ctrlutils/utils.go

```diff
@@ -141,3 +146,31 @@ func CRDExists(client client.Client, gvr schema.GroupVersionResource) bool {
 	}
 	return true
 }
+
+// retrieve Kong Admin API URL from configured name/namespace service
+func RetrieveKongAdminAPIURL(ctx context.Context, KongAdminAPI string, kubeCfg *rest.Config) (string, error) {
+	namespace, name, err := util.ParseNameNS(KongAdminAPI)
+	if err != nil {
+		return "", fmt.Errorf("failed to pars kong admin api namespace and name. err %v", err)
+	}
+	CoreClient, _ := clientset.NewForConfig(kubeCfg)
+	svc, err := CoreClient.CoreV1().Services(namespace).Get(ctx, name, metav1.GetOptions{})
+	if err != nil {
+		return "", fmt.Errorf("failed to creating kubernetes client")
+	}
+	ingresses := svc.Status.LoadBalancer.Ingress
+	adminIP := ""
+	for _, ingress := range ingresses {
+		adminIP = ingress.IP
+	}
+
+	ports := svc.Spec.Ports
+	var adminPort int32
+	for _, port := range ports {
+		if port.Name == "kong-admin" {
+			adminPort = port.Port
+		}
+	}
+	kongAdminURL := fmt.Sprintf("http://%s:%d", adminIP, adminPort)
+	return kongAdminURL, nil
```

Comment:
nit: you can return the result of `fmt.Sprint` rather than store it if there's nothing else that's going to use the variable:
```suggestion
	return fmt.Sprintf("http://%s:%d", adminIP, adminPort), nil
```

Reply:
Good catch.